### PR TITLE
NMEA RTK Detection

### PR DESCRIPTION
**Pull request by farmer-martin**
Hi everyone, new to github and code for that matter hence all the commits, anyway here goes!
Adapted from pull request #4458 and tested with an Erle robotics PFXmini on Pi3.
Enables the detection of RTK and Float from an NMEA string.

Please take a look,
Martin

### CODE REVIEW

**Comment by OXINARF** [comment]
Hi Martin,

Thank you for the PR, but it doesn't follow our rules (http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html).
Since this is your first PR and the changes are simple enough, we'll get it reviewed and if it is OK, we'll fix the commits so that we can merge it cleanly.

**Comment by OXINARF** [comment]
#3602 also had an implementation for this that looked better, although not perfect.

File: libraries/AP_GPS/AP_GPS_NMEA.cpp [hunk 1]
```diff
@@ -328,7 +328,7 @@ bool AP_GPS_NMEA::_term_complete()
                     state.status        = AP_GPS::GPS_OK_FIX_3D_RTK;      //// change to float  
                     } if (_gps_status == 4) {
                     state.status        = AP_GPS::GPS_OK_FIX_3D_RTK;
-                    } else }                  
+                    } else {
```
Comment (hiro2233) [inline libraries/AP_GPS/AP_GPS_NMEA.cpp]: lol

**Comment by WickedShell** [comment]
Closing in favour of #6949

**Pull request closed without merging**